### PR TITLE
Add a job for testing Windows on vSphere with CCM support

### PR DESCRIPTION
**Pull request by Fedosin**

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
With platform none, there should be no CCM, how is this going to work?

**Comment by Fedosin** [comment]
Yep, makes sense... I'll update it

**Comment by aravindhp** [comment]
@Fedosin @JoelSpeed having jobs for each provider with and without CCM is going to get expensive. What's the strategy here? Will there be a cut over release where CCM is the only option?

/cc @openshift/openshift-team-windows-containers

**Comment by Fedosin** [comment]
@aravindhp hey! I made this job optional, so it won't start on each test run, but on demand only.

File: ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -679,6 +679,80 @@ presubmits:
...
+    decorate: true
+    labels:
+      ci-operator.openshift.io/cloud: vsphere
+      ci-operator.openshift.io/cloud-cluster-profile: vsphere
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-windows-machine-config-operator-master-vsphere-e2e-ccm-install
+    optional: true
+    rerun_command: /test vsphere-e2e-ccm-install
+    skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
```
Comment (sebsoto) [inline ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master-presubmits.yaml:698]: Was this added manually?
seems it should be removed and be added in the config file if this job is made non-optional

**Review by sebsoto — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master-presubmits.yaml (hunk 1 above)
Comment (Fedosin) [reply ci-operator/jobs/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master-presubmits.yaml]: Hmmm, I don't remember that I added this line, but it's definitely there. I regenerated it with `make jobs` and now it's gone.

**Review by aravindhp — COMMENTED** [review]
Thanks for adding this, @Fedosin 

/lgtm

**Pull request merged**